pints/_log_pdfs.py

```diff
@@ -115,6 +115,178 @@ def sample(self, n=1):
         raise NotImplementedError
 
 
+class PooledLogPDF(LogPDF):
+    r"""
+    Creates an object of :class:LogPDF, with a log-pdf which is the sum of the
+    individual log-pdfs, while pooling selected parameters.
+
+    This :class:`LogPDF` requires that all :class:`LogPDF` objects have the
+    same number of parameters.
+
+    This is useful for e.g. modelling the time-series of multiple individuals
+    (each individual defines a separate :class:`LogPDF`), and some parameters
+    are expected to be the same across individuals (for example, the noise
+    parameter across different individuals within the same experiment).
+
+    For two :class:`LogPDF` objects :math:`L _1`  and :math:`L _2` with
+    parameters :math:`(\psi _1, \theta _1`) and :math:`(\psi _2, \theta _2`)
+    respectively, a pooling of the :math:`\theta _i` results in a pooled
+    log-pdfs of the form
+
+    .. math::
+        \log L(\psi _1, \psi _2, \theta | D_1, D_2) =
+            \log L(\psi _1, \theta | D_1) + \log L(\psi _2, \theta | D_2),
+
+    where :math:`\theta := \theta _1 = \theta _2`, and $D_i$ is the measured
+    time-series of individual :math:`i`.
+
+    In general, the search of a :class:`PooledLogPDF` has dimensionality
+    :math:`nd_{\text{up}} + d_{\text{p}}`, where :math:`n` is the number of
+    individuals, :math:`d_{\text{up}}` is the dimension of the unpooled
+    :math:`psi` parameters, and :math:`d_{\text{up}}` is the dimension of the
+    pooled :math:`\theta` parameter.
+
+    The order of parameters for evaluation is largely kept, and the parameters
+    of individual log-pdfs are concatenated. However, pooled parameters are
+    always prepended. Consider for example two log-pdf with parameters
+    :math:`(\psi _{1,1}, \psi _{1, 2}, \psi _{1,3}, \psi _{1,4})` and
+    :math:`(\psi _{2,1}, \psi _{2, 2}, \psi _{2,3}, \psi _{2,4})`. Pooling the
+    first and the last parameter, i.e.
+    :math:`\theta _1 := \psi _{1,1} = \psi _{2, 1}` and
+    :math:`\theta _4 := \psi _{1,4} = \psi _{2, 4}` results in and expect
+    order of parameters
+    :math:`(\psi _{1, 2}, \psi _{1,3}, \psi _{2, 2}, \psi _{2,3}, \theta _1,
+    \theta _4)`.
+
+    Extends :class:`LogPDF`.
+
+    Parameters
+    ----------
+    log_pdfs
+        A sequence of :class:`LogPDF` objects.
+    pooled
+        An array-like object of dtype bool, indicating which parameters across
```

Comment:
If we're going to say they need a `dtype`, then just say it _must_ be a `numpy.ndarray` (because they have an argument called `dtype`, so that's highly numpy-specific!)

Reply:
Would it be ok to say "array-like object of Booleans ... " I'd just like to clarify that this is a boolean mask to specify the pooling.

Reply:
I guess a "sequence of booleans" then?